crypto/zkp/primitives/commitments/damgard-fujisaki_square_test.go

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017 XLAB d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package commitmentzkp
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/xlab-si/emmy/crypto/commitments"
+	"github.com/xlab-si/emmy/crypto/common"
+)
+
+// TestProveDamgardFujisakiCommitmentSquare demonstrates how to prove that the commitment
+// hides the square. Given c, prove that c = g^(x^2) * h^r (mod n).
+func TestProveDamgardFujisakiCommitmentSquare(t *testing.T) {
+	receiver, err := commitments.NewDamgardFujisakiReceiver(1024, 80)
```

Comment:
I know this probably won't be possible, but still - there's quite a lot pre-initialization logic for configuring receiver and committer, which are subsequently be passed as arguments to `NewDFCommitmentSquareProver` and `NewDFCommitmentSquareVerifier`; is it in any way possible (or does it make sense at all) to initialize committer/receiver within these functions?

The same comment goes for `NewDFCommitmentEqualityProver` and `NewDFCommitmentEqualityVerifier`.

Reply:
It somehow depends on the way this proofs will be used. Let's wait with this until I implement the new scheme - I might then change this in the way you proposed (if possible).